===================================================================
RCS file: /cvsroot/mailutils/cvsroot/mailutils/mailutils/imap4d/fetch.c,v
retrieving revision 1.47
retrieving revision 1.48
diff -u -r1.47 -r1.48
--- mailutils/mailutils/imap4d/fetch.c	2003/07/26 11:07:04	1.47
+++ mailutils/mailutils/imap4d/fetch.c	2005/05/13 09:38:56	1.48
@@ -1155,6 +1168,10 @@
 	}
       else
 	util_send (" \"\"");
+    }
+  else if (end + 2 < end) /* Check for integer overflow */
+    {
+      return RESP_BAD;
     }
   else
     {
